packages/flutter/lib/src/painting/borders.dart

```diff
@@ -30,16 +30,91 @@ enum BorderStyle {
 /// - [inside] provides padding with full [BorderSide.width].
 /// - [center] provides padding with half [BorderSide.width].
 /// - [outside] provides zero padding, as stroke is drawn entirely outside.
-enum StrokeAlign {
+@immutable
+class StrokeAlign {
+  /// This constructor allows any value between 0.0 and 1.0.
+  const StrokeAlign(this.value)
+      : assert(value != null),
+        assert(value >= 0.0, 'StrokeAlign must be greater than or equal to 0.0.'),
```

Comment:
```suggestion
        assert(value >= 0.0, 'StrokeAlign must be greater than or equal to 0.0. $value was supplied instead.'),
```

It's always good to put the value that failed in the assert.